File: apps/web/src/types/alerts.ts
export type TelegramTestMessageResult = {
  status: string;
  message: string;
};

export type AlertStatus = "active" | "triggered" | "resolved" | "cancelled" | "expired";

export type AlertDeliveryStatus = "pending" | "sent" | "failed" | "skipped";

export type AlertSource = "manual" | "tradingview_webhook" | "system";

export type AlertType =
  | "info"
  | "watchlist"
  | "armed"
  | "near_trigger"
  | "entry_trigger"
  | "management"
  | "exit_warning"
  | "exit_signal"
  | "invalidation";

export type AlertEvent = {
  id: number;
  user_id: number;
  signal_id: number | null;
  trade_id: number | null;
  watchlist_item_id: number | null;
  alert_type: AlertType;
  status: AlertStatus;
  source: AlertSource;
  priority: string;
  trigger_level: string | null;
  timeframe: "1W" | "1D" | "4H" | null;
  message: string | null;
  source_payload: Record<string, unknown> | unknown[] | null;
  delivery_status: AlertDeliveryStatus;
  delivery_error: string | null;
  last_triggered_at: string | null;
  created_at: string;
  updated_at: string;
};

File: apps/api/app/schemas/alerts.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel

from app.models.enums import (
    AlertDeliveryStatus,
    AlertSource,
    AlertStatus,
    AlertType,
    NotificationChannel,
    Timeframe,
)


class AlertRead(BaseModel):
    id: int
    user_id: int
    signal_id: int | None
    trade_id: int | None
    watchlist_item_id: int | None
    alert_type: AlertType
    status: AlertStatus
    source: AlertSource
    priority: str
    trigger_level: Decimal | None
    timeframe: Timeframe | None
    message: str | None
    source_payload: dict | list | None
    delivery_status: AlertDeliveryStatus
    delivery_error: str | None
    last_triggered_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class NotificationLogRead(BaseModel):
    id: int
    user_id: int
    alert_id: int | None
    channel: NotificationChannel
    recipient: str | None
    message: str
    status: AlertDeliveryStatus
    error_message: str | None
    provider_payload: dict | list | None
    sent_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}
